pkg/util/contextutil/timeout_error.go

```diff
@@ -91,14 +100,23 @@ func decodeTimeoutError(
 		return nil
 	}
 	op := m.Details[0]
-	dur, decodeErr := time.ParseDuration(m.Details[1])
+	timeout, decodeErr := time.ParseDuration(m.Details[1])
 	if decodeErr != nil {
 		// Not encoded by our encode function. Bail out.
 		return nil //nolint:returnerrcheck
 	}
+	var took time.Duration
+	if len(m.Details) >= 3 {
+		took, decodeErr = time.ParseDuration(m.Details[2])
+		if decodeErr != nil {
+			// Not encoded by our encode function. Bail out.
+			return nil //nolint:returnerrcheck
```

Comment:
Are there any mixed-version concerns with this? I assume not since timeout errors are purely informational, but still.